pkg/service/schema.go

```diff
@@ -0,0 +1,83 @@
+package service
+
+// ParameterSchemas is a set of optional JSONSchemas that describe
+// the expected parameters for creation and update of instances and
+// creation of bindings.
+type ParameterSchemas struct {
+	ServiceInstances *InstanceSchema `json:"service_instance,omitempty"`
+	ServiceBindings  *BindingSchema  `json:"service_binding,omitempty"`
+}
+
+// InstanceSchema represents a plan's schemas for creation and
+// update of an API resource.
+type InstanceSchema struct {
+	Create *InputParameters `json:"create,omitempty"`
+	Update *InputParameters `json:"update,omitempty"`
+}
+
+// BindingSchema represents a plan's schemas for the parameters
+// accepted for binding creation.
+type BindingSchema struct {
+	Create *InputParameters `json:"create,omitempty"`
+}
+
+// InputParameters represents a schema for input parameters for creation or
+// update of an API resource.
+type InputParameters struct {
+	Parameters *ParametersSchema `json:"parameters,omitempty"`
+}
```

Comment:
I think it might be easier to understand this if this were broken up into a few different types-- `ProvisioningParametersSchema`, `UpdatingParametersSchema`, and `BindingSchema`. Such naming makes things more tractable vis a vis existing broker types.

Reply:
This is partially an artifact of the expected json that needs to get generated for the OSB spec: